src/enemy_mechanics.rs

```diff
@@ -51,3 +51,13 @@ pub fn move_flame_spirit(mut query: Query<(&mut Velocity, &mut GlobalTransform,
         }
     }
 }
+
+pub fn poison_entities(mut query: Query<(&mut Health, &mut PoisonDelay, Entity)>) {
+    for (mut health, mut poison, _) in query.iter_mut() {
+        if !poison.finished() {
+            if poison.tick_poison(Instant::now()) {
```

Comment:
It is not good idea to use Instant::now for time. add time: Res<Time> as argument and use it instead

Reply:
I don't see which methods of Time could be used here that return Instant, since the logic behind it compares Instants and putting Res<Instant> doesn't work.